src/registry/AvatarRegistry.sol

```diff
@@ -0,0 +1,523 @@
+        if (bytes(name).length == 0) {
+            revert EmptyString();
+        }
+        if (IAvatar(avatarAddress).keeper() != address(CL_REGISTRY)) {
+            revert NotCLKeeperSet();
+        }
+
+        _avatars.add(avatarAddress);
+        avatarsInfo[avatarAddress] = AvatarInfo({
+            name: name,
+            gasLimit: gasLimit,
+            status: AvatarStatus.TESTING,
+            upKeepId: _registerUpKeep(avatarAddress, gasLimit, name)
+        });
+
+        emit NewAvatar(avatarAddress, name, gasLimit, block.timestamp);
+    }
+
+    /// @dev Cancels an avatar upkeep job
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address to be cancel upkeep
+    function cancelAvatarUpKeep(address avatarAddress) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+        // NOTE: only avatar which upkeep is being cancelled can be removed
+        uint256 upKeepId = avatarsInfo[avatarAddress].upKeepId;
+        CL_REGISTRY.cancelUpkeep(upKeepId);
+    }
+
+    /// @dev Updates status of an avatar in the registry
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address update status of
+    /// @param newStatus latest status of the avatar
+    function updateStatus(address avatarAddress, AvatarStatus newStatus) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+
+        AvatarStatus oldStatus = avatarsInfo[avatarAddress].status;
+        if (oldStatus == newStatus) {
+            revert UpdateSameStatus();
+        }
+
+        avatarsInfo[avatarAddress].status = newStatus;
+
+        emit UpdateAvatarStatus(avatarAddress, oldStatus, newStatus, block.timestamp);
+    }
+
+    /// @dev Withdraws LINK funds and remove avatar from registry
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address to be remove from registry
+    function withdrawLinkFundsAndRemoveAvatar(address avatarAddress) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+
+        uint256 upKeepId = avatarsInfo[avatarAddress].upKeepId;
+        // NOTE: only avatar which upkeep is being cancelled can be removed
+        (,,,,,, uint64 maxValidBlocknumber,) = CL_REGISTRY.getUpkeep(upKeepId);
+        // https://etherscan.io/address/0x02777053d6764996e594c3e88af1d58d5363a2e6#code#F1#L738
+        if (maxValidBlocknumber == type(uint64).max) {
+            revert UpKeepNotCancelled(upKeepId);
+        }
+
+        // NOTE: removal actions after on-chain checkups
+        _avatars.remove(avatarAddress);
+        delete avatarsInfo[avatarAddress];
+
+        CL_REGISTRY.withdrawFunds(upKeepId, address(this));
+
+        emit RemoveAvatar(avatarAddress, upKeepId, block.timestamp);
+    }
```

Comment:
my plan for the keeperkeeper is to make this function part of the check/performUpkeep.

is there a reason you make this a manual step?